test/github.spec.ts

```diff
@@ -57,6 +57,36 @@ describe("Github", async() => {
   after(async() => {
     await mockAgent.close();
   });
+  test("should throw an error if the GITHUB_TOKEN environment variable is not defined", async() => {
+    const token = process.env.GITHUB_TOKEN;
+
+    process.env.GITHUB_TOKEN = "";
+
+    try {
+      assert.rejects(
+        async() => new Github("NodeSecure"),
+        {
+          name: "Error",
+          message: "Unauthorized"
+        }
+
+      );
+    }
+    finally {
+      process.env.GITHUB_TOKEN = token;
+    }
+  });
+
+  test("should generate an error when creating a Github instance with an invalid organization name", async() => {
+    assert.rejects(
```

Comment:
You don't know in the constructor if the organization is found, you need to make the request first.
You can intercept the request and mock a 404.
```js
mockPool.intercept({ path: "/orgs/foobar/repos" }).reply(404, {}, {
  headers: { "Content-Type": "application/json" }
}).persist();
```
And test it like this
```js
const fooBarGH = new Github("foobar");

assert.rejects(
  async() => await fooBarGH.fetchRepositories(),
  { message: "Not Found" }
);
```